### ISSUE TITLE
SklearnMetric integration

### ISSUE THREAD

**Issue by Scitator** [catalyst-team/catalyst#1183]
## 🚀 Feature Request

Let's add official support for all sklearn metrics to our Metric API.

### Motivation

Sklearn metrics play an essential role in ml models benchmarking.

### Proposal

Possible usage

```python
import torch
from torch.utils.data import DataLoader, TensorDataset
from catalyst import dl

# sample data
num_samples, num_features, num_classes = int(1e4), int(1e1), 4
X = torch.rand(num_samples, num_features)
y = (torch.rand(num_samples, num_classes) > 0.5).to(torch.float32)

# pytorch loaders
dataset = TensorDataset(X, y)
loader = DataLoader(dataset, batch_size=32, num_workers=1)
loaders = {"train": loader, "valid": loader}

# model, criterion, optimizer, scheduler
model = torch.nn.Linear(num_features, num_classes)
criterion = torch.nn.BCEWithLogitsLoss()
optimizer = torch.optim.Adam(model.parameters())
scheduler = torch.optim.lr_scheduler.MultiStepLR(optimizer, [2])

# model training
runner = dl.SupervisedRunner(
    input_key="features", output_key="logits", target_key="targets", loss_key="loss"
)
runner.train(
    model=model,
    criterion=criterion,
    optimizer=optimizer,
    scheduler=scheduler,
    loaders=loaders,
    logdir="./logdir",
    num_epochs=3,
    valid_loader="valid",
    valid_metric="accuracy",
    minimize_valid_metric=False,
    verbose=True,
    callbacks=[
        dl.BatchTransformCallback(
            transform=torch.sigmoid,
            scope="on_batch_end",
            input_key="logits",
            output_key="scores"
        ),
        dl.AUCCallback(input_key="logits", target_key="targets"),
        dl.MultilabelAccuracyCallback(input_key="logits", target_key="targets", threshold=0.5),
        # here is the trick
        dl.SklearnCallback(
            keys={"y_score": "scores", "y_true": "targets"}, 
            metirc_fn=sklearn.metrics.coverage_error, 
            metric_key="coverage_error",
        )
    ]
)
```

We also should add sklearn as an extra module to the Catalyst [like `ml` one](https://github.com/catalyst-team/catalyst/blob/master/catalyst/settings.py#L154).

### Alternatives

We could implement everything with PyTorch, which we should do next. Nevertheless, sklearn full support is still required as a great integration.

### Checklist
- [x] feature proposal description
- [x] motivation
- [x] extra proposal context / proposal alternatives review
  

### FAQ
Please review the FAQ before submitting an issue:
- [x] I have read the [documentation and FAQ](https://catalyst-team.github.io/catalyst/)
- [x] I have reviewed the [minimal examples section](https://github.com/catalyst-team/catalyst#minimal-examples)
- [x] I have checked the [changelog](https://github.com/catalyst-team/catalyst/blob/master/CHANGELOG.md) for main framework updates
- [x] I have read the [contribution guide](https://github.com/catalyst-team/catalyst/blob/master/CONTRIBUTING.md)
- [x] I have joined [Catalyst slack (#__questions channel)](https://join.slack.com/t/catalyst-team-core/shared_invite/zt-d9miirnn-z86oKDzFMKlMG4fgFdZafw) for issue discussion

**Comment by y-ksenia** [comment]
I could handle it:)

**Comment by Atharva-Phatak** [comment]
@Scitator How do you plan on handling the input types to sklearn metrics ? . AFAIK sklearn accepts inputs in specific shapes(mostly 1D arrays) with some post processing. I was wondering if its worth the effort or just write our own metrics in pure torch ?

**Comment by Scitator** [comment]
@Atharva-Phatak I think, we could use [BatchTransformCallback](https://catalyst-team.github.io/catalyst/api/callbacks.html#batchtransformcallback) to prepare the tensors for the metric – use BatchTransformCallback + SklearnCallback together.

**Comment by Atharva-Phatak** [comment]
@Scitator Oh wow yes that's a solution, but from a user's perspective, I would not like to write transforms and rather let the library handle the transforms. What do you think? We can write something like fastai [AccumMetrics](https://github.com/fastai/fastai/blob/master/fastai/metrics.py#L25), just an opinion. 😬

**Comment by Scitator** [comment]
🤔  looks like we have one - [AccumulationMetric](https://catalyst-team.github.io/catalyst/api/metrics.html#accumulationmetric). You could use it to store the incoming input and write your metric above the stored values.

Speaking about the Metric API – due to a large number of supported metrics Catalyst Metric API works on the most comfortable input/output for metric computation. All other preparations you could do with BatchTransformCallbacks or directly during custom `handle_batch`. That's why the current API relies on `scores`, `labels`, etc rather than `logits`, which you have to process in the metric directly (too many possible preprocessing).

**Comment by Scitator** [comment]
merged
